client/pages/conversation.jsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import Redirect from '../components/redirect';
+import AppContext from '../lib/app-context';
+import formatDate from '../lib/format-date';
+
+export default class Conversation extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      messages: null,
+      currentMessage: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSend = this.handleSend.bind(this);
+    this.refresh = this.refresh.bind(this);
+  }
+
+  refresh() {
+    const { otherId } = this.props;
+    const { userId } = this.context.user;
+    fetch(`/api/message/convo/${userId}/${otherId}`)
+      .then(response => response.json())
+      .then(messages => this.setState({ messages }));
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSend(event) {
+    event.preventDefault();
+    if (!this.state.currentMessage) {
+      return null;
+    }
+    const { currentMessage } = this.state;
+    const { otherId, postId } = this.props;
+    const { username, userId } = this.context.user;
+    const header = new Headers();
+    header.append('Content-Type', 'application/json');
+    let body = {
+      senderId: userId,
+      senderName: username,
+      recipientId: otherId,
+      content: currentMessage,
+      postId
+    };
+    body = JSON.stringify(body);
+    const init = {
+      method: 'POST',
+      headers: header,
+      body: body
+    };
+    fetch('/api/messages', init)
+      .then(response => response.json())
+      .then(message => {
+        this.setState({ currentMessage: '' }, this.refresh);
+      });
+  }
+
+  renderView() {
+    const { userId } = this.context.user;
+    const { messages } = this.state;
+    if (!messages) {
+      return (
+        <div className="row no-messages">
+          <h2>You have no messages.</h2>
+        </div>
+      );
+    }
+    return (
+      <div className="conversation-container">
+        <MessageList messages={messages} userId={userId} />
+        <form onSubmit={this.handleSend} className="row row-2 conversation-write-container">
+          <div className="convo-input shadow col-2">
+            <div className="search-icon">
+              <label htmlFor="convo-message-input"><i className="text-shadow orange fas fa-comment"></i></label>
+            </div>
+            <input autoFocus autoComplete="off" onChange={this.handleChange} type="text" className="lora" name='currentMessage' value={this.state.currentMessage} id="convo-message-input" placeholder="send a message" />
+            <button type="submit" className="send-message-button" ><i className="text-shadow fas fa-arrow-circle-up"></i></button>
+          </div>
+        </form>
+      </div>
+    );
+  }
+
+  render() {
+    if (!this.context.user) return <Redirect to="sign-in" />;
+    if (!this.state.messages) {
+      this.refresh();
```

Comment:
Instead of triggering this in `render()` it's better to "poll" the back end. Add a `componentDidMount()` method that fetches messages on an interval. Be sure to also include a `componentWillUnmount()` method that cancels the interval if the user leaves this page.